flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontend.java

```diff
@@ -110,10 +110,6 @@ public class CliFrontend {
 
     private final Options customCommandLineOptions;
 
-    private final Duration clientTimeout;
-
-    private final int defaultParallelism;
-
```

Comment:
Hi @Myracle ,
-p specifies parallelism and it is not the default parallelism.
```java
int parallelism = programOptions.getParallelism();
if (ExecutionConfig.PARALLELISM_DEFAULT == parallelism) {
    parallelism = getDefaultParallelism(effectiveConfiguration);
}
```
If parallelism is not specified (it will use the default value), then Flink will read the default value from the comfiguration file or dynamic properties.
I think we need to move a variable from a class field to a local variable if it is not required to be accessed from multiple methods.
Correct me if I am wrong.

Reply:
Hi @Myracle ，
Should I rollback so that parallelism.default could only be changed by configuration file, and dynamic properties take no effect to it?

Reply:
I think it makes sense to also fix the `defaultParallelism` in this PR.